🐛 fix: fix reading of semver for kube version

rsmitty opened the pull request:
**What this PR does / why we need it**:

This PR updates how we parse the kubernetes version in the
instances.go and allows it to pass the semver check during machine creation.

Signed-off-by: Spencer Smith <robertspencersmith@gmail.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #156

rsmitty commented:
@vincepri This should be closer to what you were looking for. Still allows for input of `v1.x.y` in the yaml files.

jdef commented on cloud/services/compute/instances.go:
```diff
@@ -61,7 +62,11 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*compute.Instance,
 			scope.Name(), scope.Namespace())
 	}
 
-	version, err := semver.Make(*scope.Machine.Spec.Version)
```
Instead of manual parsing, maybe use `ParseTolerant` (instead of `Make`).

rsmitty commented:
@jdef @vincepri okay, updated to use parse tolerant. feels much cleaner, thanks for the pointer to that function! also added in the `v` for the default source image.
